src/VisualStudio/LiveShare/Impl/Shims/GoToDefinitionHandlerShim.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.LiveShare.LanguageServices;
+
+namespace Microsoft.VisualStudio.LanguageServices.LiveShare.Shims
+{
+    internal class GoToDefinitionHandlerShim : AbstractLiveShareHandlerShim<TextDocumentPositionParams, object>
```

Comment:
what is a 'handler shim'?

Reply:
is this type ever instantiated directly?  if not, make abstract.

Reply:
the shims export the required liveshare type, but use the LSP implementation defined in a lower layer (instead of a custom liveshare impl).  Ideally, the entire liveshare implementations would just be shims, but everything with a UI dependency cannot be moved.